portal-kernel/src/com/liferay/portal/kernel/upgrade/BaseExternalReferenceCodeUpgradeProcess.java

```diff
@@ -96,7 +96,7 @@ protected void upgradeExternalReferenceCode(
 						connection, updateSB.toString())) {
 
 				while (resultSet.next()) {
-					long primKey = resultSet.getLong(1);
```

Comment:
If you do that, you will break the other upgrade classes that are also using this base class. So we can't change the current behavior of this class. If you want, you should adjust it to give support to non primary key columns.